python/ray/serve/tests/test_logging.py

```diff
@@ -578,5 +585,52 @@ def format_and_verify_json_output(record, expected_record: dict):
     format_and_verify_json_output(record, expected_json)
 
 
+@pytest.mark.parametrize(
+    "log_encoding",
+    [
+        [None, "TEXT"],
+        ["TEXT", "TEXT"],
+        ["JSON", "JSON"],
+        ["FOOBAR", "TEXT"],
+    ],
+)
+def test_configure_component_logger_with_log_encoding_env(monkeypatch, log_encoding):
```

Comment:
please also test the override behavior if users specify the encoding in their `logging_config`. I assume that env var is lowest precedence

Reply:
Makes sense, but this kinda requires us to default to an encoding that's neither text nor json in the `logging_config` so we know if user has explicitly set it. Let me take a stab

Reply:
Right now just use `None` to represent the encoding not being set in the logging config. But open to adding new enum of `NOT_SET` or other ideas :)

Reply:
Ouch, this actually won't work. The protobuf doesn't like `None`😅 Gonna just add a `NOT_SET` enum